components/shared/ResourceTag.tsx

```diff
@@ -0,0 +1,56 @@
+import React from "react";
+import workflowBadge from "@/public/icons/monitor.svg";
+import knowledgeBadge from "@/public/icons/greenbubble.svg";
+import componentBadge from "@/public/icons/numberlist.svg";
+import Image from "next/image";
+type ResourceTagType = "knowledge" | "component" | "workflow" | "plain";
+interface CustomBadgeProps extends React.PropsWithChildren {
+  type: ResourceTagType;
```

Comment:
```suggestion
  type?: ResourceTagType;
```
If we make this optional, we can short-circuit the value in our class and have a "default" class if you don't want to set that value.